### PR TITLE
Update to make use of some Java 8 features

### PR DESCRIPTION
**Pull request by rachelriv**
* Modified the word count program to show off some of the functional-style operations available on streams in Java 8
* Small fix in `TripleApp.java`

@rtoal @alexschneider Would be curious to know what you guys think.

### CODE REVIEW

File: java/WordCountApp.java [hunk 1]
```diff
@@ -1,26 +1,19 @@
...
-                counts.put(word, counts.getOrDefault(word, 0) + 1);
-            }
-        }
-        for (Map.Entry<String, Integer> e : counts.entrySet()) {
-            System.out.printf("%s %d\n", e.getKey(), e.getValue());
-        }
+        ((new BufferedReader(new InputStreamReader(System.in)))
+            .lines()
+            .flatMap(line -> Arrays.stream(line.split("[^a-zA-Z']+")))
+            .collect(groupingBy(String::toLowerCase, TreeMap::new, counting())))
```
Comment (rtoal) [inline java/WordCountApp.java]: +1 for flatMap, nice work

File: java/WordCountApp.java [hunk 2]
```diff
@@ -1,26 +1,19 @@
...
-                word = word.toLowerCase();
-                counts.put(word, counts.getOrDefault(word, 0) + 1);
-            }
-        }
-        for (Map.Entry<String, Integer> e : counts.entrySet()) {
-            System.out.printf("%s %d\n", e.getKey(), e.getValue());
-        }
+        ((new BufferedReader(new InputStreamReader(System.in)))
+            .lines()
+            .flatMap(line -> Arrays.stream(line.split("[^a-zA-Z']+")))
```
Comment (rtoal) [inline java/WordCountApp.java]: I think you have to lower case first. How does this work with "Word" and "word"?

**Comment by alexschneider** [comment]
LGTM

**Comment by jadengore** [comment]
Java 8 :sparkles:

**Pull request merged**